avro-serializer/src/main/java/io/confluent/kafka/serializers/KafkaAvroDecoder.java

```diff
@@ -28,16 +28,20 @@ public KafkaAvroDecoder(SchemaRegistryClient schemaRegistry) {
     this.schemaRegistry = schemaRegistry;
   }
 
-  public KafkaAvroDecoder(SchemaRegistryClient schemaRegistry, VerifiableProperties props) {
+  public KafkaAvroDecoder(
+          SchemaRegistryClient schemaRegistry,
+          VerifiableProperties props,
+          boolean isKey) {
+
     this.schemaRegistry = schemaRegistry;
-    configure(deserializerConfig(props.props()));
+    configure(deserializerConfig(props.props()), isKey);
```

Comment:
We should leave the constructor as-is and just pass null as the `type` to `configure`